### ISSUE TITLE
Could not find gem ' * ruby' in the gems available on this machine.

### ISSUE THREAD

**Issue by Ch4s3** [bundler/bundler#3380]
I recently updated to the latest version, 1.7.12 and when I change a gem version in my gemfile and run `bundle` or `bundle install` I get a message like:
````
Fetching gem metadata from https://rails-assets.org/...
Fetching gem metadata from https://rails-assets.org/.
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies...
Could not find gem 'typhoeus (~> 0.7.1) ruby' in the gems available on this machine.
````

In this example I changed the following:
```` gem "typhoeus", "~> 0.7.0" ````

to:
```` gem "typhoeus", "~> 0.7.1" ````

**Comment by brockspratlen** [comment]
@Ch4s3 does your Gemfile have multiple ```source``` definitions? We are seeing a similar issue when we have a Gemfile similar to the following:

```
source "https://rubygems.org"

source "https://geminabox.example.com" do
  gem "some_private_gem"
end

gem "geocoder"
```

We haven't been able to reproduce the issue when the only ```source``` defined in our Gemfile is ```https://rubygems.org```. Curious if it is the same for you.

**Comment by brockspratlen** [comment]
actually, nevermind. I missed the references to ```https://rails-assets.org``` in the sample output you posted. That tells me you've got an extra ```source``` for rails-assets. I think you're seeing the same problem that we reported in #3377.

Could you share your Gemfile? Feel free to remove anything that you feel is private, really just curious about the lines that start with ```source```.

**Comment by Ch4s3** [comment]
I'm out of town, but I'll put up a slightly edited version Monday.

**Comment by TimMoore** [comment]
This should be fixed in 1.7.13. Please try upgrading and let us know if you still have any problems.
